filesystem/iso9660.ksy

```diff
@@ -5,269 +5,625 @@ meta:
       - id: magic
-        contents: "CD001"
+        doc-ref: ecma-119 8.1.2
+        contents: [0x43, 0x44, 0x30, 0x30, 0x31]
       - id: version
-        type: u1
-      - id: vol_desc_boot_record
-        type: vol_desc_boot_record
-        if: type == 0
-      - id: vol_desc_primary
-        type: vol_desc_primary
-        if: type == 1
-  vol_desc_boot_record:
+        doc-ref: ecma-119 8.1.3
+        contents: [0x01]
+      - id: volume
+        type:
+          switch-on: descriptor_type
+          cases:
+            'descriptor_type::boot_record_volume_descriptor': boot_record_volume
+            'descriptor_type::primary_volume_descriptor': primary_volume
+            'descriptor_type::supplementary_volume_descriptor': supplementary_volume
+#            'descriptor_type::volume_partition_descriptor': volume_partition
+  boot_record_volume:
+    doc-ref: ecma-119 8.2
+    seq:
+      - id: boot_system_identifier
+        doc-ref: ecma-119 8.2.4
+        type: strz
+        size: 0x20
+        encoding: ASCII
+      - id: boot_identifier
+        doc-ref: ecma-119 8.2.5
+        type: strz
+        size: 0x20
+        encoding: ASCII
+  datetime_long:
+    doc-ref: ecma-119 8.4.26.1
     seq:
-      - id: boot_system_id
+      - id: year
         type: str
-        size: 32
-        encoding: UTF-8
-      - id: boot_id
+        size: 0x4
+        encoding: ASCII
+      - id: month
         type: str
-        size: 32
-        encoding: UTF-8
-  vol_desc_primary:
-    doc-ref: 'http://wiki.osdev.org/ISO_9660#The_Primary_Volume_Descriptor'
+        size: 0x2
+        encoding: ASCII
+      - id: day
+        type: str
+        size: 0x2
+        encoding: ASCII
+      - id: hour
+        type: str
+        size: 0x2
+        encoding: ASCII
+      - id: minute
+        type: str
+        size: 0x2
+        encoding: ASCII
+      - id: second
+        type: str
+        size: 0x2
+        encoding: ASCII
+      - id: hundredths_second
+        type: str
+        size: 0x2
+        encoding: ASCII
+      - id: timezone_offset
+        type: s1
+  primary_volume:
+    doc-ref: ecma-119 8.4
     seq:
-      - id: unused1
-        contents: [0]
-      - id: system_id
+      - id: unused01
+        doc-ref: ecma-119 8.4.4
+        contents: [ 0x0 ]
+      - id: system_identifier
+        doc-ref: ecma-119 8.4.5
         type: str
-        size: 32
-        encoding: UTF-8
-      - id: volume_id
+        size: 0x20
+        encoding: ASCII
+      - id: volume_identifier
+        doc-ref: ecma-119 8.4.6
         type: str
-        size: 32
-        encoding: UTF-8
-      - id: unused2
-        contents: [0, 0, 0, 0, 0, 0, 0, 0]
-      - id: vol_space_size
+        size: 0x20
+        encoding: ASCII
+      - id: unused02
+        doc-ref: ecma-119 8.4.7
+        contents: [ 0x0, 0x0, 0x0, 0x0,  0x0, 0x0, 0x0, 0x0 ]
+      - id: volume_space_size
+        doc-ref: ecma-119 8.4.8
         type: u4bi
-      - id: unused3
```

Comment:
```reserved``` is IMHO a better name.

Reply:
use ```-orig-id``` for the original name (and it doesn't need to be a valid KS id) and use the most convenient in ```id```

Reply:
I am sorry, but I am using 4, maybe 6 standards soon, adding my own names just makes a bigger mess. I am willing to change a lot, this one I will not.

Reply:
Completely OK, it's your work, do as you think is better, just keep in mind that you are allowed and encouraged rename things.